File: app/src/main/java/com/treelzebub/umap/ui/activity/LoginActivity.java
package com.treelzebub.umap.ui.activity;

import android.app.Activity;
import android.app.AlertDialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.webkit.WebView;
import android.widget.Button;
import android.widget.EditText;

import com.idiogram.umap.R;

/**
 * Created by Tre Murillo on 3/23/15
 */
public class LoginActivity extends Activity {

    public static final String TAG = "LoginActivity";

    public boolean mHasLoggedIn = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        if (!mHasLoggedIn) {
            AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
            LayoutInflater inflater = this.getLayoutInflater();
            View dialogView = inflater.inflate(R.layout.dialog_oauth, null);
            dialogBuilder.setView(dialogView);

            WebView webView = (WebView) dialogView.findViewById(R.id.webview);
            EditText authCodeET = (EditText) dialogView.findViewById(R.id.auth_code_et);
            Button submitButton = (Button) dialogView.findViewById(R.id.submit_button);

            webView.loadUrl("http://www.asdf.com");
        }

    }
}

File: app/src/main/java/net/treelzebub/umap/ui/LoginActivity.kt
package net.treelzebub.umap.ui

import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.webkit.WebView
import android.webkit.WebViewClient
import butterknife.bindView
import com.squareup.otto.Subscribe
import net.treelzebub.umap.Constants
import net.treelzebub.umap.R
import net.treelzebub.umap.async.event.AccessTokenEvent
import net.treelzebub.umap.auth.AuthService
import net.treelzebub.umap.auth.LoginUtils
import net.treelzebub.umap.auth.TokenHolder
import net.treelzebub.umap.util.BusProvider
import net.treelzebub.umap.util.async

/**
 * Created by Tre Murillo on 5/28/15
 *
 * Provides a one-time login to Discogs.com
 */
public class LoginActivity : AppCompatActivity() {

    val webView: WebView by bindView(R.id.webview)

    private var authUrl: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        BusProvider.instance.register(this)
        setContentView(R.layout.activity_login)
        loadAuthUrl()
        val wvSettings = webView.settings
        wvSettings.builtInZoomControls = true
        wvSettings.javaScriptEnabled = true
        webView.setWebViewClient(RequestTokenCallback())
    }

    private fun loadAuthUrl() {
        async({
            val sAuth = AuthService.instance
            val rt = sAuth.requestToken
            TokenHolder.requestToken = rt
            authUrl = sAuth.getAuthorizationUrl(rt) + Constants.DISCOGS_AUTH_URL_APPEND + rt.token
        }, {
            webView.loadUrl(authUrl)
        })
    }

    private inner class RequestTokenCallback() : WebViewClient() {
        override fun shouldOverrideUrlLoading(view: WebView, url: String?): Boolean {
            if (url != null && url.startsWith(Constants.CALLBACK_URL)) {
                //TODO spinner visible
                LoginUtils.requestAccessToken(this@LoginActivity, Uri.parse(url))
                return true
            }
            return false
        }
    }

    @Subscribe
    public fun onAccessToken(e: AccessTokenEvent) {
        //TODO spinner gone
        startActivity(DashboardActivity.getIntent(this@LoginActivity))
    }
}
